### PR TITLE
ISSUE-172 - support basic authentication for apigee api calls

### LINKED ISSUE #172: Basic Auth support for Apigee integration
We need agents to support Basic Auth and http for integration with apigee management Api. Oauth2 is not enabled by default for Edge for the private Cloud.

Sample api request to fetch list of api products :
curl --location 'http://{host}:{port}/v1/organizations/{org_name}/apiproducts' \
--header 'Accept: application/json' \
--header 'Authorization: Basic ABCDEFGHxxxxx=='

Reference document to Apigee Edge Management APIs: https://apidocs.apigee.com/operations
There are sample requests and responses provided by Apigee which can be referenced in the link. Agents should be able to adapt to these api's in order for successful integration.

We would be able to integrate agents and Private Cloud Apigee with ease.

### PR DESCRIPTION
**Pull request by jcollins-axway**
Fixes #172 

- Additionally add some unit tests

### CODE REVIEW

**Review by sbolosan — APPROVED** [review]

**Review by vivekschauhan — APPROVED** [review]

**Review by sbolosan — APPROVED** [review]

**Review by vivekschauhan — APPROVED** [review]

**Comment by jcollins-axway** [comment]
@alrosca, @dgghinea please review

File: client/pkg/config/authconfig.go [hunk 1]
```diff
@@ -33,3 +34,8 @@ func (a *AuthConfig) GetUsername() string {
 func (a *AuthConfig) GetPassword() string {
 	return a.Password
 }
+
+// GetPassword - Returns true if Basic Authentication should be used for the APIGEE api calls
```
Comment (dgghinea) [inline client/pkg/config/authconfig.go:38]: I believe this should say UseBasicAuth instead of GetPassword

File: client/pkg/config/config.go [hunk 2]
```diff
@@ -107,11 +119,12 @@ func AddProperties(rootProps properties.Properties) {
 	rootProps.AddStringProperty(pathAuthURL, "https://login.apigee.com", "URL to use when authenticating to APIGEE")
 	rootProps.AddStringProperty(pathAuthServerUsername, "edgecli", "Username to use to when requesting APIGEE token")
 	rootProps.AddStringProperty(pathAuthServerPassword, "edgeclisecret", "Password to use to when requesting APIGEE token")
+	rootProps.AddStringProperty(pathAuthUsername, "", "Username to use to authenticate to APIGEE")
+	rootProps.AddStringProperty(pathAuthPassword, "", "Password for the user to authenticate to APIGEE")
+	rootProps.AddBoolProperty(pathAuthBasicAuth, false, "Set to tru to use basic authentication to authenticate to APIGEE")
```
Comment (dgghinea) [inline client/pkg/config/config.go:124]: small typo "tru"

File: client/pkg/config/authconfig.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ type AuthConfig struct {
 	ServerPassword string `config:"serverPassword"`
 	Username       string `config:"username"`
 	Password       string `config:"password"`
+	BasicAuth      bool   `config:"useBasicAuth"`
```
Comment (dgghinea) [inline client/pkg/config/authconfig.go:10]: Should this be "HasBasicAuth" or "UseBasicAuth" ?

**Review by dgghinea — APPROVED** [review]

File: client/pkg/config/authconfig.go (hunk 3 above)
Comment (jcollins-axway) [reply client/pkg/config/authconfig.go:10]: The attribute is just basic auth, the getter is UseBasicAuth, I think keeping it this way is fine. Thanks for the suggestion.

**Pull request merged**